### PR TITLE
hwmv2: Port STM32F3X SoCs & boards

### PR DESCRIPTION
**Pull request by ajarmouni-st**
Port STM32F3X SoCs & boards to Hardware Model V2.

### CODE REVIEW

File: boards/st/stm32373c_eval/stm32373c_eval_defconfig [hunk 1]
```diff
@@ -1,11 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Zephyr Kernel Configuration
-CONFIG_SOC_SERIES_STM32F3X=y
-
 # Platform Configuration
-CONFIG_SOC_STM32F373XC=y
-
```
Comment (erwango) [inline boards/st/stm32373c_eval/stm32373c_eval_defconfig:8]: Comments to be cleaned up

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ajarmouni-st** [comment]
Rebased to solve conflicts & address comments above.

**Comment by nordicjm** [comment]
rebase needed

File: boards/st/stm32f3_disco/revision.cmake
Comment (erwango) [inline boards/st/stm32f3_disco/revision.cmake:1]: This file could be removed.

File: boards/st/nucleo_f334r8/nucleo_f334r8_defconfig [hunk 2]
```diff
@@ -1,11 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Zephyr Kernel Configuration
-CONFIG_SOC_SERIES_STM32F3X=y
-
 # Platform Configuration
```
Comment (erwango) [inline boards/st/nucleo_f334r8/nucleo_f334r8_defconfig:4]: Obsolete comments should be removed.

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

**Comment by nordicjm** [comment]
Rebase needed

**Comment by ajarmouni-st** [comment]
> Rebase needed

Done

**Pull request merged**